File: src/main/java/frc/team4069/robot/commands/elevator/SetElevatorPositionCommand.java
package frc.team4069.robot.commands.elevator;

import frc.team4069.robot.commands.CommandBase;
import frc.team4069.robot.subsystems.ElevatorSubsystem.Position;

// Use our enum values with MotionMagic to move the elevator to predefined locations passed in the constructor
public class SetElevatorPositionCommand extends CommandBase {
    private Position position;

    public SetElevatorPositionCommand(Position position) {
        this.position = position;
        requires(elevator);
    }

    @Override
    protected void initialize() {
        elevator.setPosition(position);
    }

    @Override
    protected boolean isFinished() {
        return true;
    }
}

File: src/main/java/frc/team4069/robot/commands/elevator/SetElevatorPositionCommand.kt
package frc.team4069.robot.commands.elevator

import edu.wpi.first.wpilibj.command.InstantCommand
import frc.team4069.robot.subsystems.ElevatorSubsystem

// A command to instruct the elevator to move to a predefined vertical position
class SetElevatorPositionCommand(private val position: ElevatorSubsystem.Position) : InstantCommand() {

    // Require the elevator subsystem
    init {
        requires(ElevatorSubsystem)
    }

    // Set the elevator's position to whatever was passed to this
    override fun initialize() {
        ElevatorSubsystem.setPosition(position)
    }
}